app/daemon.go

```diff
@@ -408,6 +416,20 @@ func (d *MenderShellDaemon) setupLogging() {
 	}
 }
 
+func (d *MenderShellDaemon) DecreaseSpawnedShellsCount(shellStoppedCount uint) {
+	d.spawnedShellsMutex.Lock()
+	defer d.spawnedShellsMutex.Unlock()
+	if d.shellsSpawned == 0 {
+		log.Warn("can't decrement shellsSpawned count: it is 0.")
+	} else {
+		if shellStoppedCount >= d.shellsSpawned {
+			d.shellsSpawned = 0
+		} else {
+			d.shellsSpawned -= shellStoppedCount
+		}
+	}
+}
```

Comment:
I believe you can use `sync/atomic` for this operation as well :)

Reply:
aha! that I wanted to use for some time. thanks. I promise -- next time. (and it was discussed privately)